device.go

```diff
@@ -203,12 +216,9 @@ func (d *Device) SetAutoDetach(autodetach bool) error {
 	if d.handle == nil {
 		return fmt.Errorf("SetAutoDetach(%v) called on %s after Close", autodetach, d)
 	}
-	var autodetachInt int
-	switch autodetach {
-	case true:
-		autodetachInt = 1
-	case false:
-		autodetachInt = 0
+	d.autodetach = autodetach
+	if autodetach && d.detached == nil{
+		d.detached = make(map[int]bool)
```

Comment:
I'm not sure if that's really necessary. After the configuration is switched, the kernel will reattach whatever drivers it considers necessary for the new configuration. After that, autoattach will only detach the claimed interfaces, and reattach them automatically, all withinh libusb.